OpenIGTLinkIF/Logic/vtkSlicerOpenIGTLinkIFLogic.cxx

```diff
@@ -690,8 +715,87 @@ void vtkSlicerOpenIGTLinkIFLogic::GetDeviceNamesFromMrml(IGTLMrmlNodeListType &l
     }
 }
 
-////---------------------------------------------------------------------------
-//const char* vtkSlicerOpenIGTLinkIFLogic::MRMLTagToIGTLName(const char* mrmlTagName);
-//{
-//
-//}
+//---------------------------------------------------------------------------
+void vtkSlicerOpenIGTLinkIFLogic::vtkInternal::UpdateNodeLocator(vtkMRMLTransformNode* transformNode)
+{
+  // Set visibility of attached model node
+  bool visible = (transformNode->GetAttribute("IGTLVisible") &&
+    strcmp(transformNode->GetAttribute("IGTLVisible"), "true") == 0);
+  vtkMRMLModelNode* locatorModel = vtkMRMLModelNode::SafeDownCast(
+    transformNode->GetNodeReference(this->External->GetLocatorModelReferenceRole()));
+  if (!locatorModel && visible)
+    {
+    if (this->External->GetMRMLScene())
+      {
+      std::stringstream ss;
+      ss << "Locator_" << transformNode->GetName();
+      std::string locatorString = ss.str();
+      locatorModel = this->AddLocatorModel(this->External->GetMRMLScene(), ss.str(), 0.0, 1.0, 1.0);
+      if (locatorModel)
+        {
+        locatorModel->SetAndObserveTransformNodeID(transformNode->GetID());
+        transformNode->SetNodeReferenceID(this->External->GetLocatorModelReferenceRole(), locatorModel->GetID());
+        }
+      }
+    }
+  if (locatorModel)
+    {
+    locatorModel->SetDisplayVisibility(visible);
+    }
+}
+
+//---------------------------------------------------------------------------
+vtkMRMLModelNode* vtkSlicerOpenIGTLinkIFLogic::vtkInternal::AddLocatorModel(vtkMRMLScene * scene, std::string nodeName, double r, double g, double b)
+{
+
+  vtkSmartPointer<vtkMRMLModelNode> locatorModel = vtkSmartPointer<vtkMRMLModelNode>::New();
+  vtkSmartPointer<vtkMRMLModelDisplayNode> locatorDisp = vtkSmartPointer<vtkMRMLModelDisplayNode>::New();
+
+  // Cylinder represents the locator stick
+  vtkSmartPointer<vtkCylinderSource> cylinder = vtkSmartPointer<vtkCylinderSource>::New();
+  cylinder->SetRadius(1.5);
+  cylinder->SetHeight(100);
+  cylinder->SetCenter(0, 0, 0);
+  cylinder->Update();
```

Comment:
no need to call Update() if you connect filters using SetInputConnection/GetOutputPort